### DISCUSSION TITLE
When is the "delete" method in Auth Persistance Layer (APL) called?

### DISCUSSION THREAD

**Discussion by madmonkeyworks** [saleor/saleor#14465] [Q&A]
Hi there, 

I'm writing a custom Auth persistence layer (APL). The "set" method fires correctly and saves the authData, however, I wonder what's the idea behind the "delete" method. When is "delete" called? Can you shed some light on that? 

I'm looking for a way to delete the authData when an app is deleted from Saleor. Do I need to implement the APP_DELETED webhook?
Regards

**Comment by witoszekdev** [comment]
Hi, @madmonkeyworks currently this APL method isn't called, due to an issue in Saleor with `APP_DELETED` webhook. From the APL perspective you don't need to do anything, once the issue is resolved the auth data should be removed from APL on app uninstall.

Here's the link to the issue: https://github.com/saleor/saleor/issues/12298

**Reply by madmonkeyworks** [reply]
Hi @witoszekdev, I understand that, for now, I will just leave the authData in the database when the app gets deleted - no problem. 
Can you describe the mechanism of how will the APL remove the authData when the issue is solved? Will the api/register be called upon APP_DELETED event? Or will we still have to make a dedicated api endpoint for app deleted event?

**Reply by lkostrowski** [reply]
Hey, most likely it will be a dedicated endpoint. Either a webhook or endpoint registered in app manifest
